### ISSUE TITLE
Two node templates can't use the same Rancher Managed NSG

### ISSUE THREAD

**Issue by aiyengar2** [rancher/tasks#70]
TLDR: two node templates using the same Rancher managed NSG name will "compete" over the Open Ports assigned to the underlying NSG. ~If a user wants to refer to the same underlying NSG, one of the node templates should refer to it as an existing NSG.~ (see comment below, a Rancher managed NSG cannot be shared across node templates)

Issue:

Currently, upon creating a node template, a Rancher managed NSG name is populated in the 
NSG field as shown in the screenshot below:

![Screen Shot 2020-06-10 at 11 53 38 AM](https://user-images.githubusercontent.com/13127443/84307008-22829b80-ab11-11ea-9402-e21a61826f02.png)

Assuming a user were to create a node template `test-nsg-on-azure-1` with this NSG `rancher-managed-Doiw1zh1` and the following Open Ports, they would be able to provision a node successfully with these ports opened up on the NSG:
`6443/tcp,2379/tcp,2380/tcp,8472/udp,4789/udp,9796/tcp,10256/tcp,10250/tcp,10251/tcp,10252/tcp`

If they then update the Open Ports (i.e. add `80/tcp, 443/tcp` to the list) and provision another node, the underlying NSG will be updated and both nodes will have the new list of Open Ports applied to them.

However, if a user were to then create a node template `test-nsg-on-azure-2` which has a different default Rancher managed NSG name (i.e. `rancher-managed-Gsfg4dh2`), intentionally replace that with the existing Rancher managed NSG name they have (i.e. `rancher-managed-Doiw1zh1`), and then have a different set of Open Ports listed on their node template, the underlying NSG `rancher-managed-Doiw1zh1` will be updated and the nodes that were initially spun up from `test-nsg-on-azure-1` will be impacted by the changed Open Port rules. Similarly, if another node is later spun off from `test-nsg-on-azure-1`, the nodes provisioned by `test-nsg-on-azure-2` will be impacted by the changed Open Port rules from `test-nsg-on-azure-1`.

~Quick Mitigation:~ (see comment below, this will not resolve the underlying issue)

~Can we add a note onto the help text that says something along the lines of "If using an existing NSG or an NSG referenced in another node template, please provide the full ARM resource identifier for the NSG."?~

It might be a good idea to introduce some kind of API level validation to address this requirement on node template update.

**Comment by aiyengar2** [comment]
In regard to API-level validation to prevent this, we could introduce a check on updating Azure node templates in Rancher. If the Azure node template's NSG field is non-null and does not contain any slashes (i.e. a pooled NSG), check to see that no other node template has this value in their NSG field.

**Comment by aiyengar2** [comment]
Looking into the `rancher/machine` code, the mitigation suggested above actually doesn't resolve the problem: https://github.com/rancher/machine/blob/f675ca219acb7402720f060432bc81e9032f6e85/drivers/azure/azureutil/azureutil.go#L142-L144

Regardless of the method of input (i.e. as an ARM resource identifier or a normal string), a NSG tagged with `rancher-managed` will have its rules overridden by a node template that references it. If a user would like to use one NSG across multiple node templates, they should use an existing NSG.

Alternatively, they can remove the tag from the Rancher Managed NSG on the Azure portal (effectively converting it into an existing NSG) at which point they can reference it directly. However, in this case, future changes to the Open Ports on Azure node templates will not be honored and Rancher will not automatically delete the NSG when it is no longer being used.
